pkg/dataplane/reloadCredntials.go

```diff
@@ -0,0 +1,117 @@
+package dataplane
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/msi-dataplane/pkg/dataplane/swagger"
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-logr/logr"
+)
+
+type reloadingCredential struct {
+	currentValue *azidentity.ClientCertificateCredential
+	cloud        string
+	lock         *sync.RWMutex
+	logger       *logr.Logger
+}
+
+type Option func(*reloadingCredential)
+
+func WithLogger(logger logr.Logger) Option {
+	return func(c *reloadingCredential) {
+		c.logger = &logger
+	}
+}
+
+func NewUserAssignedIdentityCredential(ctx context.Context, cloud string, credentialPath string, opts ...Option) (azcore.TokenCredential, error) {
+	credential := &reloadingCredential{
+		cloud: cloud,
+		lock:  &sync.RWMutex{},
```

Comment:
Maybe let's add a default logger here so we still spit out the errors if users don't configure something in particular.

Reply:
(Silently failing without any indication is a frustrating experience for users)

Reply:
We can use the stdlib logger so we don't pull in other dependencies.